compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/lower/MoveCompanionObjectFieldsLowering.kt

```diff
@@ -83,9 +84,15 @@ private class MoveCompanionObjectFieldsLowering(val context: CommonBackendContex
         companion.declarations.removeAll { it is IrAnonymousInitializer }
     }
 
-    private fun IrClass.allFieldsAreJvmField() =
-        declarations.filterIsInstance<IrProperty>()
-            .mapNotNull { it.backingField }.all { it.hasAnnotation(JVM_FIELD_ANNOTATION_FQ_NAME) }
+    private fun shouldExposeProperty(irClass: IrClass, irProperty: IrProperty): Boolean {
+        if (!irClass.isInterface && !irClass.isAnnotationClass)
+            return true
+
+        return irProperty.backingField?.let {
+            it.hasAnnotation(JVM_FIELD_ANNOTATION_FQ_NAME) || it.correspondingPropertySymbol?.owner?.isConst == true ||
+                    it.correspondingPropertySymbol?.owner?.isLateinit == true
```

Comment:
I don't see any tests with lateinit properties. Is this really required? In general, could you please add a comment to this method explaining why only these particular cases should be moved?

Reply:
I followed [the doc](https://kotlinlang.org/docs/reference/java-to-kotlin-interop.html#static-fields) blindly. They are probably not needed as it doesn't make sense for interface fields, which are `public static final`, to be lateinit.

Reply:
Why did you remove the `isLateinit == true` condition in the latest version? Yes for interfaces it doesn't make sense, but you can have a `lateinit` property in a companion object for non-interface classes.

Reply:
Yes, `lateinit` properties will be copied for non-interfaces classes. They are enabled by the first condition `!irClass.isJvmInterface`. I've updated the comments.